tests/integration/support/client_wrapper.go

```diff
@@ -0,0 +1,216 @@
+package support
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/optimizely/go-sdk/pkg/client"
+	"github.com/optimizely/go-sdk/pkg/config"
+	"github.com/optimizely/go-sdk/pkg/decision"
+	"github.com/optimizely/go-sdk/pkg/entities"
+	"github.com/optimizely/go-sdk/pkg/event"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+	"github.com/optimizely/go-sdk/tests/integration/optlyplugins"
+	"gopkg.in/yaml.v3"
+)
+
+// ClientWrapper - wrapper around the optimizely client that keeps track of various custom components used with the client
+type ClientWrapper struct {
+	Client          *client.OptimizelyClient
+	DecisionService decision.Service
+	EventDispatcher event.Dispatcher
+}
+
+// NewClientWrapper returns a new instance of the optly wrapper
+func NewClientWrapper(datafileName string) ClientWrapper {
+
+	datafileDir := os.Getenv("DATAFILES_DIR")
+	datafile, err := ioutil.ReadFile(filepath.Clean(path.Join(datafileDir, datafileName)))
+	if err != nil {
+		log.Fatal(err)
+	}
+	configManager, err := config.NewStaticProjectConfigManagerFromPayload(datafile)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	eventProcessor := event.NewEventProcessor(event.BatchSize(models.EventProcessorDefaultBatchSize), event.QueueSize(models.EventProcessorDefaultQueueSize), event.FlushInterval(models.EventProcessorDefaultFlushInterval))
+
+	optimizelyFactory := &client.OptimizelyFactory{
+		Datafile: datafile,
+	}
+
+	decisionService := &optlyplugins.TestCompositeService{CompositeService: *decision.NewCompositeService("")}
+	eventProcessor.EventDispatcher = &optlyplugins.ProxyEventDispatcher{}
+
+	client, err := optimizelyFactory.Client(
+		client.WithConfigManager(configManager),
+		client.WithDecisionService(decisionService),
+		client.WithEventProcessor(eventProcessor))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return ClientWrapper{
+		Client:          client,
+		DecisionService: decisionService,
+		EventDispatcher: eventProcessor.EventDispatcher,
+	}
+}
+
+// InvokeAPI processes request with arguments
+func (c *ClientWrapper) InvokeAPI(request models.APIOptions, response *models.APIResponse) error {
```

Comment:
I think we should have `InvokeAPI` return the `(APIResponse, error)` instead of receiving one to populate. Same for the individual requests, --> `c.isFeatureEnabled` should return the APIResponse